### ISSUE TITLE
Memory Usage with SQ Higher Than Memory Usage Without

### ISSUE THREAD

**Issue by robertgshaw2-neuralmagic** [vllm-project/llm-compressor#79]
**Describe the bug**
For `Llama-70B-Instruct`, I run OOM on 8xA100 running `SmoothQuant` and `GPTQ` but not when only running `GPTQ`. The OOM occurs during the `GPTQ` phase (so `SmoothQuant` itself runs fine).

**Expected behavior**
I would expect similar memory usage

**Environment**
8xA100

**To Reproduce**
- commenting / uncommenting the `SmoothQuant` line:
```python
from datasets import load_dataset
from transformers import AutoTokenizer

from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.modifiers.smoothquant import SmoothQuantModifier
from llmcompressor.transformers import SparseAutoModelForCausalLM, oneshot

MODEL_ID = "meta-llama/Meta-Llama-3-70B-Instruct"
SAVE_DIR = MODEL_ID.split("/")[1] + "-W8A8-Dynamic"

# 1) Load model (device_map="auto" with shard the model over multiple GPUs!).
model = SparseAutoModelForCausalLM.from_pretrained(
    MODEL_ID,
    device_map="auto",
    torch_dtype="auto",
)
tokenizer = AutoTokenizer.from_pretrained(MODEL_ID)

# 2) Prepare calibration dataset (in this case, we use ultrachat).
DATASET_ID = "HuggingFaceH4/ultrachat_200k"
DATASET_SPLIT = "train_sft"

# Select number of samples. 512 samples is a good place to start.
# Increasing the number of samples can improve accuracy.
NUM_CALIBRATION_SAMPLES = 512
MAX_SEQUENCE_LENGTH = 1024

# Load dataset and preprocess.
ds = load_dataset(DATASET_ID, split=DATASET_SPLIT)
ds = ds.shuffle(seed=42).select(range(NUM_CALIBRATION_SAMPLES))


def preprocess(example):
    return {
        "text": tokenizer.apply_chat_template(
            example["messages"],
            tokenize=False,
        )
    }


ds = ds.map(preprocess)


# Tokenize inputs.
def tokenize(sample):
    return tokenizer(
        sample["text"],
        padding=False,
        max_length=MAX_SEQUENCE_LENGTH,
        truncation=True,
        add_special_tokens=False,
    )


ds = ds.map(tokenize, remove_columns=ds.column_names)

# 3) Configure algorithms. In this case, we:
#   * apply SmoothQuant to make the activations easier to quantize
#   * quantize the weights to int8 with GPTQ (static per channel)
#   * quantize the activations to int8 (dynamic per token)
recipe = [
    # SmoothQuantModifier(smoothing_strength=0.8),
    GPTQModifier(targets="Linear", scheme="W8A8", ignore=["lm_head"]),
]

# 4) Apply algorithms and save in `compressed-tensors` format.
oneshot(
    model=model,
    tokenizer=tokenizer,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    output_dir=SAVE_DIR,
)

```

**Errors**
If applicable, add a full print-out of any errors or exceptions that are raised or include screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here. Also include any relevant files.

**Comment by Satrat** [comment]
Hey @robertgshaw2-neuralmagic this should be fixed with this commit: https://github.com/vllm-project/llm-compressor/commit/453022a185555e46423028c2fe3efc71f9cf66ca 

The issue was we were allocating some additional memory during SmoothQuant scaling and it wasn't being Freed. As for the uneven GPU allocation you noticed, I confirmed this is expected. When the model is first loaded, we evenly split it across all GPUs (while ensuring no decoder layers are split). For larger amounts of GPUs this results in the first few GPUs containing most of the decoder layers, while the final GPU contains the lm_head and fewer decoder layers. Since we don't apply GPTQ and allocate hessians for the lm_head, we see less memory usage on that GPU.

**Comment by robertgshaw2-neuralmagic** [comment]
> Hey @robertgshaw2-neuralmagic this should be fixed with this commit: [453022a](https://github.com/vllm-project/llm-compressor/commit/453022a185555e46423028c2fe3efc71f9cf66ca)
> 
> The issue was we were allocating some additional memory during SmoothQuant scaling and it wasn't being Freed. As for the uneven GPU allocation you noticed, I confirmed this is expected. When the model is first loaded, we evenly split it across all GPUs (while ensuring no decoder layers are split). For larger amounts of GPUs this results in the first few GPUs containing most of the decoder layers, while the final GPU contains the lm_head and fewer decoder layers. Since we don't apply GPTQ and allocate hessians for the lm_head, we see less memory usage on that GPU.

That makes sense.

I will retry my workflow with the commit. Thanks!
